packages/client/tests/server/handlers.ts

```diff
@@ -592,6 +596,78 @@ export const handlers = [
         );
     }),
 
+    rest.post<SignUpRequestBody, Record<string, never>, SignUpResponseBody>(
+        `${SERVER_ENDPOINT}/authentication/sign-up`,
+        (req, res, ctx) => {
+            const signUpFailureReasonCollection: SignUpFailureReasons[] = [];
+            const { authenticationMode, email, password, userNickname } =
+                req.body;
+
+            if (!z.string().max(255).email().check(email)) {
+                signUpFailureReasonCollection.push('INVALID_EMAIL');
+            }
+
+            const userWithSameNickname = db.authUser.findFirst({
+                where: {
+                    nickname: {
+                        equals: userNickname,
+                    },
+                },
+            });
+            if (userWithSameNickname) {
+                signUpFailureReasonCollection.push('UNAVAILABLE_NICKNAME');
+            }
+
+            const userWithSameEmail = db.authUser.findFirst({
+                where: {
+                    email: {
+                        equals: email,
+                    },
+                },
+            });
+            if (userWithSameEmail) {
+                signUpFailureReasonCollection.push('UNAVAILABLE_EMAIL');
+            }
+
+            if (!passwordStrengthRegex.test(password)) {
+                signUpFailureReasonCollection.push('WEAK_PASSWORD');
+            }
+
+            if (signUpFailureReasonCollection.length > 0) {
+                return res(
+                    ctx.status(400),
+                    ctx.json({
+                        signUpFailureReasonCollection,
+                        status: 'FAILURE',
+                    }),
+                );
+            }
+
+            const userID = datatype.uuid();
+            db.authUser.create({
+                userID,
+                email,
+                password,
+                nickname: userNickname,
+            });
+
+            const isApiTokenAuthentication = authenticationMode === 'api';
+            return res(
+                ctx.status(200),
+                ctx.json({
+                    status: 'SUCCESS',
+                    userSummary: {
+                        nickname: userNickname,
+                        userID,
+                    },
+                    token: isApiTokenAuthentication
+                        ? datatype.uuid()
+                        : undefined,
+                }),
+            );
+        },
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏Having a server becomes useless when you have MSW. 👌